vendor/github.com/docker/swarmkit/manager/allocator/cnmallocator/drivers_ipam.go

```diff
@@ -14,24 +13,24 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-func initIPAMDrivers(r *drvregistry.DrvRegistry, defaultAddrPool []*net.IPNet, subnetSize int) error {
+func initIPAMDrivers(r *drvregistry.DrvRegistry, netConfig *NetworkConfig) error {
 	var addressPool []*ipamutils.NetworkToSplit
 	var str strings.Builder
 	str.WriteString("Subnetlist - ")
 	// Extract defaultAddrPool param info and construct ipamutils.NetworkToSplit
 	// from the info. We will be using it to call Libnetwork API
 	// We also need to log new address pool info whenever swarm init
 	// happens with default address pool option
-	if defaultAddrPool != nil {
-		for _, p := range defaultAddrPool {
+	if netConfig != nil {
+		for _, p := range netConfig.DefaultAddrPool {
 			addressPool = append(addressPool, &ipamutils.NetworkToSplit{
-				Base: p.String(),
-				Size: subnetSize,
+				Base: p,
+				Size: int(netConfig.SubnetSize),
```

Comment:
Pity that we still need the conversion from `uint32` to `int` here; I assume changing `ipamutils.NetworkToSplit` will be problematic?